src/rockthevote.cpp

```diff
@@ -0,0 +1,494 @@
+/**
+ * =============================================================================
+ * CS2Fixes
+ * Copyright (C) 2023 Source2ZE
+ * =============================================================================
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, version 3.0, as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rockthevote.h"
+#include "commands.h"
+#include "playermanager.h"
+#include "ctimer.h"
+#include "entity/cgamerules.h"
+
+#include "tier0/memdbgon.h"
+
+extern CEntitySystem *g_pEntitySystem;
+extern IVEngineServer2 *g_pEngineServer2;
+extern CGlobalVars *gpGlobals;
+extern CCSGameRules *g_pGameRules;
+
+ERTVState g_RTVState = ERTVState::MAP_START;
+EExtendState g_ExtendState = EExtendState::MAP_START;
+int g_ExtendsLeft = 1;
+
+//TODO POSSIBLE CONVARS
+float g_RTVSucceedRatio = 0.6f;
+float g_ExtendSucceedRatio = 0.5f;
+int g_ExtendTimeToAdd = 20;
+
+#define ADMIN_PREFIX "Admin %s has "
```

Comment:
I would move this to adminsystem.h and include that file, also remove the definition in adminsystem.cpp.